_posts/2022-10-09-decentralization-of-zk-rollups.md

```diff
@@ -0,0 +1,192 @@
+---
+layout: post
+author: Ibrahim Yusufali
+title: Decentralization of ZK Rollups
+excerpt: "Zero knowledge rollups were used in the past couple of years to solve the scalability constraints in L1 monolithic chains. A lot of new projects have emerged in the space since then, each of them ambitious and hoping to be the “ultimate” solution. However, as it is with every rapidly growing technology there are certain aspects that may be overlooked, in order to facilitate fast execution and entry to the market. We will be doing a deep dive into what we believe is one of those aspects: “decentralization”."
+---
+
+_Many thanks to Preston Evans, Shashank Agrawal, Ventali Tan, Daniel Lubarov, Gautam Botrel and James Stearn for feedback._
+
+## Outline
+
+1. Introduction
+2. Upgrade Keys
+3. Why Proof of Stake (PoS) Might Not Work
+4. Proof of Efficiency
+5. Proof Generation Outsourcing + MEV Auction
+6. Our Proposals
+7. Conclusion
+
+
+## Introduction
+
+Zero knowledge rollups were used in the past couple of years to solve the scalability constraints in L1 monolithic chains. A lot of new projects have emerged in the space since then, each of them ambitious and hoping to be the “ultimate” solution. However, as it is with every rapidly growing technology there are certain aspects that may be overlooked, in order to facilitate fast execution and entry to the market. We will be doing a deep dive into what we believe is one of those aspects: “decentralization”. 
+
+All current zk rollups are centralized. Some do not even have a projected path to decentralization. This is not a good situation to be in. Some features in a truly decentralized zk-rollup which we hope to achieve:
```

Comment:
Is this claim true `All current zk rollups are centralized`? Can the term `centralized` be defined first, and can the strong claim holds such that no project can challenge its validity? Do we have a list of projects that do not have a "projected path to decentralization"? This needs to be prepared and made sure to hold 100% when challenged.

Do we have a reference or sources for this list of features? Is there some balance and trade-offs between these features, and how are projects doing with respect to each feature at this time? In absence of these, the term "truly" seems subjective and may make readers think we do not believe any project is on the path of becoming "truly decentralized"

Reply:
The claim that "All current zk rollups are centralized" is in fact true and cannot be disputed. Centralized in my minds means mainly the decentralization of the sequencer. I will make this explicit in the article. None of the polygon sub-suites, Scroll, zk-Sync, Starkware or Aztec have implemented sequencer decentralization yet. To my knowledge, zkSync, Starkware and Aztec are the two projects that yet to publish a path to decentralization. 

Regarding the term "truly decentralized", I have evaluated Polygon Hermez and Scroll's progress against these features in the following paragraphs (and these are the only two that have published plans on decentralizaiton afaik). The features were collected mainly from the implementation plan introductions, and evaluating the direction of discussions around the topic. See https://jumpcrypto.com/a-framework-for-analyzing-l1s/, https://hackmd.io/@yezhang/SkmyXzWMY, https://ethresear.ch/t/proof-of-efficiency-a-new-consensus-mechanism-for-zk-rollups/11988. I agree that this might be controversial, but I have made sure to check this with the reviewers, and it has been iterated and expanded based on their feedback.

Reply:
add to footnote?